### ISSUE TITLE
Improvements for alternative color-less mode

### ISSUE THREAD

**Issue by regexident** [regexident/cargo-modules#2]
`cargo-modules` in its current and initial release makes use of ANSI terminal colors for distinguishing between different "types" of modules. This unfortunately not only [alienates Windows users](https://github.com/regexident/cargo-modules/issues/1) (due to a lack of Windows support by the [colored crate](https://crates.io/crates/colored)), and those with a dark-on-light terminal theme, but even worse: it alienates color-blind users. That's unacceptable to me.

---

There are currently four different colors used for distinguishing between:

- test modules
- public modules
- private modules
- orphaned modules

How would you for an alternative plain display mode prefer the tree to be presented?

Please feel free to leave comments and/or alternative suggestions in the comments.

---

There are basically three usage scenarios to cover here:

- user has terminal **with** color support and **prefers colored** tree.
- user has terminal **with** color support and **prefers plain** tree.
- user has terminal **without** color support and has no choice.

So looking at this I'm wondering if colored output should actually be made opt-in via `--colors`?

**Comment by regexident** [comment]
Would something like this work for you?

```
phrases (public)
 ├── english (public)
 │   ├── greetings (public)
 │   │   └── test (test)
 │   └── farewells (public)
 │       └── test (test)
 ├── japanese (private)
 │   ├── greetings (private)
 │   └── farewells (private)
 └── latin (orphaned)
```

👍 ? 👎 ?

**Comment by ejmahler** [comment]
It's worth pointing out that colorblind users don't necessarily prefer a colorless tree -- simply that there needs to be secondary and maybe tertiary ways to distinguish them. think of things like a plus sign to the left of the name for public, and a minus sign for private, in addition to the color.

**Comment by matthieu-m** [comment]
As mentioned by ejmahler, removing colors altogether is not necessary. Also, it should be mentioned that color-blindness is actually a fairly generic term referring both to persons have difficulty seeing some colors (I have some trouble distinguished green from maroon myself) up to persons only seeing nuances of grey. In the middle, you find people not seeing one color at all (I had a colleague who saw the red color... black, so that red on black was just plain black for him).

Annotating on top of coloring would work, it might be a bit heavyweight (verbose), but it's immediately clear and after a while I expect that non color-blind users will just follow the color code and ignore whatever is in parenthesis (the human brain is really good at filtering out uninteresting stuff).

**Comment by regexident** [comment]
I hope the just published v0.2.1 addresses your points. Would love to hear your feedback on it!

**Comment by matthieu-m** [comment]
It is much better, and the addition of "cfg" attributes is really cool to! Thanks!
